lib/options.json

```diff
@@ -145,7 +145,34 @@
           "type": "boolean"
         },
         {
-          "type": "object"
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string"
+            },
+            "host": {
+              "type": "string"
+            },
+            "port": {
+              "type": "number"
+            },
+            "type": {
+              "type": "string"
+            },
+            "protocol": {
+              "type": "string"
+            },
+            "subTypes": {
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "txt": {
+              "type": "object"
+            }
+          },
+          "additionalProperties": false
```

Comment:
I still think we don't need this, new options can be added and we don't need do extra work here it is 3rd party code

Reply:
https://github.com/webpack/webpack-dev-server/pull/3202#discussion_r616536873

Reply:
hasn't done any changes since 5y

Reply:
Tomorrow all can be changed

Reply:
lets avoid it then 😮

Reply:
done

Reply:
yes, lets keep only `boolean` and set `https` when `options.https` else `http`

Reply:
https://github.com/webpack/webpack-dev-server/pull/3202#discussion_r616536946 we need support all options

Reply:
`protocol (string, optional) - udp or tcp (default)` and `txt (object, optional) - a key/value object to broadcast as the TXT record` can be useful too in some cases